.github/workflows/publish.yml

```diff
@@ -0,0 +1,44 @@
+name: Publish Packages
+
+on:
+  workflow_dispatch:
+
+concurrency: version-or-publish-${{ github.ref }}
+
+jobs:
+  publish:
+    name: Publish Packages
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    environment: publish
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+        with:
+          fetch-depth: 0 # To get all tags
+          ref: ${{ github.ref }}
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - name: Create Prepare Release PR or Publish
+        id: changesets
+        uses: changesets/action@001cd79f0a536e733315164543a727bdf2d70aff # v1.5.1
+        with:
+          title: Prepare Release
+          commit: Prepare Release
+          version: npm run version
+          publish: npm run publish
+          commitMode: github-api
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
For future reference making comment to enable config provenance when pushing

Reply:
Adding link to reference comment for tracking: https://github.com/OpenZeppelin/contracts-wizard/pull/528#discussion_r2072026221